src/dnsmasq_interface.c

```diff
@@ -3133,15 +3117,6 @@ void FTL_TCP_worker_created(const int confd)
 	if(config.debug != 0)
 		logg("Reopening Gravity database for this fork");
 	gravityDB_forked();
-
-	// Children inherit file descriptors from their parents
-	// We don't need them in the forks, so we clean them up
-	if(config.debug != 0)
-		logg("Closing Telnet socket for this fork");
-	close_telnet_socket();
-	if(config.debug != 0)
-		logg("Closing Unix socket for this fork");
-	close_unix_socket(false);
```

Comment:
I just want to leave some details where why we don't need this in the new version. This code is called when FTL forks (creates a dedicated worker for DNS TCP requests). When a multi-threaded process forks, only the calling thread is cloned in the child process - the other threads of the father process are not forked: the child process is mono-threaded until it creates new threads on its side. However, we are only listening on the sockets in the respective threads of the main application (non-forked UDP process). There is nothing in the forked process that would be listening so we can be sure that no API requests are accidentally replied to from forks.